src/modules/hip/kernel/rain.hpp

```diff
@@ -0,0 +1,300 @@
+
+    d_float24 src1_f24, src2_f24, dst_f24;
+
+    rpp_hip_load24_pln3_and_unpack_to_float24_pkd3(srcPtr1 + srcIdx, srcStridesNCH.y, &src1_f24);
+    rpp_hip_load24_pln3_and_unpack_to_float24_pkd3(srcPtr2 + srcIdx, srcStridesNCH.y, &src2_f24);
+    rain_hip_compute(&src1_f24.f8[0], &src2_f24.f8[0], &dst_f24.f8[0], &alpha_f4);
+    rain_hip_compute(&src1_f24.f8[1], &src2_f24.f8[1], &dst_f24.f8[1], &alpha_f4);
+    rain_hip_compute(&src1_f24.f8[2], &src2_f24.f8[2], &dst_f24.f8[2], &alpha_f4);
+    rpp_hip_pack_float24_pkd3_and_store24_pkd3(dstPtr + dstIdx, &dst_f24);
+}
+
+template <typename T>
+RppStatus hip_exec_rain_tensor(T *srcPtr1,
+                               RpptDescPtr srcDescPtr,
+                               T *dstPtr,
+                               RpptDescPtr dstDescPtr,
+                               Rpp32f rainPercentage,
+                               Rpp32u rainWidth,
+                               Rpp32u rainHeight,
+                               Rpp32s slant,
+                               Rpp32f *alpha,
+                               RpptROIPtr roiTensorPtrSrc,
+                               RpptRoiType roiType,
+                               rpp::Handle& handle)
+{
+    if (roiType == RpptRoiType::LTRB)
+        hip_exec_roi_converison_ltrb_to_xywh(roiTensorPtrSrc, handle);
+
+    Rpp32f rainPercent = rainPercentage * 0.004f;
+    Rpp32u numDrops = static_cast<Rpp32u>(rainPercent * srcDescPtr->h * srcDescPtr->w);
+    std::srand(std::time(0));
+    T *rainLayer = reinterpret_cast<T *>(handle.GetInitHandle()->mem.mgpu.scratchBufferPinned.floatmem);
+    T initValue;
+    if constexpr (std::is_same<T, Rpp8s>::value)
+    {
+        initValue = static_cast<T>(0x81);
+    }
+    else
+    {
+        initValue = static_cast<T>(0);
+    }
+    hipMemset(rainLayer, initValue, srcDescPtr->strides.nStride * sizeof(T));
+    for (Rpp32u i = 0; i < numDrops; i++)
+    {
+        Rpp32u xStart = rand() % (srcDescPtr->w - slant);
+        Rpp32u yStart = rand() % (srcDescPtr->h - rainHeight);
+        for (Rpp32u j = 0; j < rainHeight; j++) // height - rainHeight
+        {
+            Rpp32u x = xStart + j * slant / rainHeight;
+            Rpp32u y = yStart + j;
+
+            if (x >= 0 && x < srcDescPtr->w && y < srcDescPtr->h)
+            {
+                Rpp32s loc = y * srcDescPtr->strides.hStride + x * srcDescPtr->strides.wStride;
+                T *rainLayerTemp = rainLayer + loc;
+
+                // Conditionally assign rain value based on type T
+                T rainValue;
+                if constexpr (std::is_same<T, Rpp8u>::value)
+                {
+                    rainValue = static_cast<T>(200);
+                }
+                else if constexpr (std::is_same<T, Rpp8s>::value)
+                {
+                    rainValue = static_cast<T>(72);
+                }
+                else
+                {
+                    rainValue = static_cast<T>(200 * ONE_OVER_255);
+                }
```

Comment:
Please refer to comments added in HOST kernel for this function

Reply:
Done